### PR TITLE
Nikita/Connecting CFD account creation to upgrade modal 

### PR DESCRIPTION
**Pull request by nikita-deriv**

### CODE REVIEW

File: packages/core/src/Stores/client-store.js [hunk 1]
```diff
@@ -736,6 +736,52 @@ export default class ClientStore extends BaseStore {
         }
     }
 
+    @action.bound
+    redirectToMt5Real = (market_type, server, doRedirect) => {
+        const synthetic_server_region = server ? server.server_info?.geolocation.region : '';
```
Comment (amina-deriv) [inline packages/core/src/Stores/client-store.js:741]: can we add optional chaining( ? ) to geolocation also?

Comment (nikita-deriv) [reply packages/core/src/Stores/client-store.js]: Thanks for advice, updated here.

File: packages/core/src/Stores/client-store.js [hunk 2]
```diff
@@ -747,6 +747,52 @@ export default class ClientStore extends BaseStore {
...
+            return !!(citizen && tax_identification_number && tax_residence);
+        };
+
+        const has_required_account = account_type === 'synthetic' ? this.has_malta_account : this.has_maltainvest_account;
+        const should_redirect_fstp_password = this.is_fully_authenticated && hasRequiredCredentials();
+
+        if (this.is_eu && !has_required_account) {
+            sessionStorage.setItem('cfd_account_needed', 1);
+            this.root_store.ui.toggleAccountsDialog(false);
+            // TO DO remove account needed modal logic
```
Comment (ali-hosseini-fs) [inline packages/core/src/Stores/client-store.js:782]: is it intended to be here?

File: packages/core/src/App/Containers/AccountSwitcher/account-switcher.jsx [hunk 3]
```diff
@@ -247,12 +201,16 @@ const AccountSwitcher = props => {
     // Real accounts is always the first tab index based on design
     const isRealAccountTab = active_tab_index === 0;
 
+    const getDemoMT5 = () => {
```
Comment (ali-hosseini-fs) [inline packages/core/src/App/Containers/AccountSwitcher/account-switcher.jsx:204]: I added this function to the client store.
You can connect to it using `demo_mt5_accounts: client.demo_mt5_accounts,`
Please just remove it and use the new one

File: packages/core/src/Stores/client-store.js (hunk 2 above)
Comment (nikita-deriv) [reply packages/core/src/Stores/client-store.js:782]: Just left it here to future improvement. Can be removed.

File: packages/core/src/App/Containers/AccountSwitcher/account-switcher.jsx (hunk 3 above)
Comment (nikita-deriv) [reply packages/core/src/App/Containers/AccountSwitcher/account-switcher.jsx:204]: Okay, thanks, will do refactor for this.

**Pull request merged**